react-search-component/my-app/src/SearchableList Components/SearchableList.tsx

```diff
@@ -0,0 +1,24 @@
+import SearchBar from './SearchBar';
+import TextDisplay from './TextDisplay';
+import { useState } from 'react';
+
+type SearchableListProps = {
+  list: { id: string; sentence: string }[];
+};
+
+export default function SearchableList({ list }: SearchableListProps) {
+  const [search, setSearch] = useState('');
+
+  function handleOnSearch(searchText: string) {
+    setSearch((prevSearchText) => {
+      return searchText;
```

Comment:
I'm not sure why you are using the function form of the state setter here. You never use `prevSearchText`. Isn't `setSearch(searchText)` simpler and works just as well?

Reply:
When I was learning the state setter and what it does, a lot of the videos in youtube kept mentioning to make it a callback function in which I find more easier to understand.